test/e2e/autoscaling_utils.go

```diff
@@ -79,30 +94,59 @@ func (rc *ResourceConsumer) makeConsumeCPURequests() {
 	var rest int
 	for {
 		select {
-		case millicores := <-rc.channel:
+		case millicores := <-rc.cpu:
 			count = millicores / requestSizeInMillicores
 			rest = millicores - count*requestSizeInMillicores
 		case <-time.After(sleepTime):
 			if count > 0 {
-				rc.sendConsumeCPUrequests(count, requestSizeInMillicores, consumptionTimeInSeconds)
+				rc.sendConsumeCPURequests(count, requestSizeInMillicores, consumptionTimeInSeconds)
+			}
+			if rest > 0 {
+				go rc.sendOneConsumeCPURequest(rest, consumptionTimeInSeconds)
+			}
+		case <-rc.stopCPU:
+			return
+		}
+	}
+}
+
+func (rc *ResourceConsumer) makeConsumeMemRequests() {
+	defer GinkgoRecover()
```

Comment:
I think we don't need GinkgoRecover here (and in makeConsumeCPUReqeusts). It is called in sendOneConsumeMem(CPU)Request and we don't call fail/assert in other places.